includes/class-theme-base.php

```diff
@@ -1496,7 +1559,73 @@ function the_pagination_html( $args = array() ) {
 	 */
 	function get_pagination_html( $args = array() ) {
 
-		echo get_the_posts_pagination( $args );
+		get_the_posts_pagination( $args );
```

Comment:
@sarutole This uses `global $wp_query` and `$wp_rewrite`.   Since one of WPLib's goal is to eliminate a lot of errors around globals, shouldn't we capture both variables into a property of $theme and use it directly here by saving, setting then restoring the global var?

We already gave a `$theme->query()` method although it just accesses global `$wp_the_query`, it does not actually capture it.  Maybe it should?   In hindsight maybe that one should have been named `$theme->main_query()` to allow for a `$theme->query()` based on `$wp_query`?